tests/integration/test_core.py

```diff
@@ -89,9 +91,64 @@ async def test_mysql_relation(
     await ops_test.model.deploy("charmed-osm-mariadb-k8s", application_name="mariadb")
     await ops_test.model.add_relation("wordpress", "mariadb:mysql")
     await ops_test.model.wait_for_idle()
+
     assert (
             await get_app_status("wordpress") == ops.model.ActiveStatus.name
     ), (
         "application status should be active once correct database connection info "
         "being provided via relation"
     )
+
+
+@pytest.mark.asyncio
+async def test_get_initial_password_action(default_admin_password):
+    """
+    arrange: after WordPress charm has been deployed
+    act: run get-initial-password action
+    assert: get-initial-password action should return default admin password
+    """
+    assert (
+            len(default_admin_password) > 8
+    ), "get-initial-password action should return the default password"
+
+
+@pytest.mark.asyncio
+async def test_wordpress_functionality(
+        ops_test: pytest_operator.plugin.OpsTest,
+        application_name,
+        default_admin_password
+):
+    """
+    arrange: after WordPress charm has been deployed and db relation established
+    act: test WordPress basic functionality (login, post, comment)
+    assert: WordPress works normally as a blog site
+    """
+    units = (await ops_test.model.get_status()).applications[application_name].units.values()
+    for unit in units:
+        unit_ip = unit.address
+        wp_client = WordpressClient(
+            host=unit_ip,
+            username="admin",
+            password=default_admin_password,
+            is_admin=True
+        )
+        post_title = secrets.token_hex(16)
+        post_content = secrets.token_hex(16)
+        post = wp_client.create_post(
+            title=post_title,
+            content=post_content,
+        )
+        homepage = wp_client.get_homepage()
+        assert (
+                post_title in homepage and post_content in homepage
+        ), "admin user should be able to create a new post"
+        comment = secrets.token_hex(16)
```

Comment:
Do you think it is worth putting a comment here explaining why a secret token is used as the comment? I'm guessing the test just needs some random string? Would some fixed string or a string including e.g., the name of the unit work?

Reply:
It's just a nonce, this test will run multiple times and each time needs a unique title and content. But I think a comment about this is a good ideal.